core/src/main/kotlin/coursebook/service/CoursebookService.kt

```diff
@@ -1,15 +1,42 @@
 package com.wafflestudio.snu4t.coursebook.service
 
+import com.wafflestudio.snu4t.common.enum.Semester
 import com.wafflestudio.snu4t.coursebook.data.Coursebook
 import com.wafflestudio.snu4t.coursebook.repository.CoursebookRepository
+import kotlinx.coroutines.flow.toList
 import org.springframework.stereotype.Service
 
 interface CoursebookService {
     suspend fun getLatestCoursebook(): Coursebook
+
+    suspend fun getCoursebooks(): List<Coursebook>
+
+    suspend fun getSyllabusUrl(year: Int?, semester: Semester?, courseNumber: String?, lectureNumber: String?): String
 }
 
 @Service
 class CoursebookServiceImpl(private val coursebookRepository: CoursebookRepository) : CoursebookService {
     override suspend fun getLatestCoursebook(): Coursebook =
         coursebookRepository.findFirstByOrderByYearDescSemesterDesc()
+
+    override suspend fun getCoursebooks(): List<Coursebook> = coursebookRepository.findAllByOrderByYearDescSemesterDesc().toList()
+    override suspend fun getSyllabusUrl(year: Int?, semester: Semester?, courseNumber: String?, lectureNumber: String?): String {
+        return """
+            http://sugang.snu.ac.kr/sugang/cc/cc103.action?openSchyy=${year ?: "undefined"}
+            &openShtmFg=${makeOpenShtmFg(semester)}&openDetaShtmFg=${makeOpenDetaShtmFg(semester)}
+            &sbjtCd=${courseNumber ?: "undefined"}&ltNo=${lectureNumber ?: "undefined"}&sbjtSubhCd=000
+        """.replace("\\s+".toRegex(), "")
```

Comment:
```kotlin
val urlBuilder = DefaultUriBuilderFactory().builder()
    .host("http://sugang.snu.ac.kr")
    .path("/sugang/cc/cc103.action")
```

이런거 대충 두고 

```suggestion
        return urlBuilder
            .queryParam("openSchyy", year)
            .queryParam("openShtmFg", makeOpenShtmFg(semester))
            .queryParam("openDetaShtmFg", makeOpenDetaShtmFg(semester))
            .queryParam("sbjtCd", courseNumber)
            .queryParam("ltNo", lectureNumber)
            .queryParam("sbjtSubhCd", "000")
            .build()
            .toString()
```

이렇게 바꾸면 좋을듯요

Reply:
지금은 좀 보기 싫네요

Reply:
오..상세한 피드백 감사합니다 :)